### PR TITLE
Provision LB resources on prepare, (re)configure on activate

### PR DESCRIPTION
**Pull request by jonmv**
@freva and @mpolden please review.

### CODE REVIEW

**Review by freva — APPROVED** [review]

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisionerTest.java [hunk 1]
```diff
@@ -221,8 +227,9 @@ public void provision_load_balancer_combined_cluster() {
         var combinedId = ClusterSpec.Id.from("container1");
         var nodes = prepare(app1, clusterRequest(ClusterSpec.Type.combined, ClusterSpec.Id.from("content1"), Optional.of(combinedId), ZoneEndpoint.defaultEndpoint));
         assertEquals(1, lbs.get().size());
-        assertEquals("Prepare provisions load balancer with reserved nodes", 2, lbs.get().get(0).instance().get().reals().size());
+        assertEquals("Prepare provisions load balancer wihtout reserved nodes", 0, lbs.get().get(0).instance().get().reals().size());
```
Comment (mpolden) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisionerTest.java:230]: without

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisioner.java [hunk 2]
```diff
@@ -345,12 +349,12 @@ private Set<String> reachableIpAddresses(Node node) {
     private static void requireInstance(LoadBalancerId id, LoadBalancer loadBalancer, CloudAccount cloudAccount, ZoneEndpoint zoneEndpoint) {
         if (loadBalancer.instance().isEmpty()) {
             // Signal that load balancer is not ready yet
-            throw new LoadBalancerServiceException("Could not (re)configure " + id + ". The operation will be retried on next deployment");
+            throw new LoadBalancerServiceException("Could not provision " + id + ". The operation will be retried on next deployment");
         }
-        if (!inAccount(cloudAccount, loadBalancer)) {
+        if ( ! inAccount(cloudAccount, loadBalancer)) {
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisioner.java:354]: Hopefully we'll decide on (and enforce) a code style some day. 🙂

**Review by mpolden — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisioner.java (hunk 2 above)
Comment (jonmv) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisioner.java:354]: 🙈Yes.

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisionerTest.java (hunk 1 above)
Comment (jonmv) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisionerTest.java:230]:
```suggestion
        assertEquals("Prepare provisions load balancer without reserved nodes", 0, lbs.get().get(0).instance().get().reals().size());
```

**Pull request merged**